src/schemas/jsonSchema/index.js

```diff
@@ -0,0 +1,395 @@
+  }
+
+  async addClaimDefinition(schema) {
+    const definition = {
+      identifier: schema.title,
+      version: getSchemaVersion(schema.title),
+      type: await this.findDefinitionType(schema),
+    };
+
+    if (definition.type === 'Array') {
+      const subSchema = await this.loadSchemaFromUri(schema.items.$ref);
+
+      definition.items = {
+        type: subSchema.title,
+      };
+    }
+
+    ['attestable', 'credentialItem', 'minimum', 'maximum'].forEach((property) => {
+      if (property in schema) {
+        definition[property] = schema[property];
+      }
+    });
+
+    if (schema.required) {
+      definition.type.required = schema.required;
+    }
+
+    this.definitions.push(definition);
+    this.validIdentifiers.push(schema.title);
+    SummaryMapper.addDefinition(definition);
+  }
+
+  async getPropertyValues(properties) {
+    const defProperties = [];
+
+    // eslint-disable-next-line guard-for-in,no-restricted-syntax
+    for (const name in properties) {
+      const property = properties[name];
+
+      let { type } = property;
+      const { items } = property;
+
+      if (type === 'array') {
+        if (items.$ref) {
+          // TODO: remove hardcode identifier
+          type = items.$ref.replace('http://identity.com/schemas/', '');
+        } else {
+          type = _.capitalize(type);
+        }
+      }
+
+      if (property.allOf) {
+        // eslint-disable-next-line no-await-in-loop
+        const schema = await this.loadSchemaFromUri(property.allOf[0].$ref);
+        type = schema.title;
+      }
+
+      const defProperty = { name, type };
+
+      defProperties.push(defProperty);
+    }
+
+    return { properties: defProperties };
+  }
+
+  /**
+   * Finds the definition/properties of a schema
+   */
+  async findDefinitionType(schema) {
+    if (schema.type === 'object') {
+      if (!_.isEmpty(schema.properties)) {
+        const values = await this.getPropertyValues(schema.properties);
+        return values;
+      }
+
+      const subSchema = await this.loadSchemaFromUri(schema.allOf[0].$ref);
+      if (subSchema == null) {
+        return null;
+      }
+
+      return subSchema.title;
+    }
+
+    if (schema.type === 'array') {
+      return 'Array';
+    }
+
+    return _.capitalize(schema.type);
+  }
+
+  /**
+   * Loads a schema, traversing all the subschemas and loading them as well
+   */
+  async loadSchemaFromTitle(title) {
+    const loader = this.findSchemaLoader(title);
+    if (loader == null) {
+      return null;
+    }
+
+    const schemaId = loader.schemaId(title);
+    const validateSchema = this.ajv.getSchema(schemaId);
+
+    let schema;
+    if (!validateSchema) {
+      schema = await loader.loadSchema(title);
+      if (schema === null) {
+        return null;
+      }
+
+      try {
+        this.ajv.addSchema(schema);
+      } catch (e) {
+        // TODO: This could only happen if we have a cyclic dependency, or the same ref multiple times in the schema...
```

Comment:
What the plan for this TODO? There is already a story for the in the backlog? should we add one?